pkg/odo/cli/preference/unset_test.go

```diff
@@ -0,0 +1,80 @@
+package preference
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/preference"
+)
+
+func TestUnsetForce(t *testing.T) {
+	tests := []struct {
+		name           string
+		forceFlag      bool
+		exists         bool
+		expectedRunErr string
+	}{
+		{
+			name:      "force && parameter exists",
+			forceFlag: true,
+			exists:    true,
+		},
+		{
+			name:           "no force and parameter not exists",
+			forceFlag:      false,
+			exists:         false,
+			expectedRunErr: "preference already unset",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			prefClient := preference.NewMockClient(ctrl)
+			opts := NewUnsetOptions(prefClient)
+			opts.forceFlag = tt.forceFlag
+
+			cmdline := cmdline.NewMockCmdline(ctrl)
+
+			args := []string{"Arg1"}
+			err := opts.Complete(cmdline, args)
+			if err != nil {
+				t.Errorf("Expected nil error, got %s", err)
+				return
+			}
+
+			if opts.paramName != "arg1" {
+				t.Errorf("Expected paramName %q, got %q", "arg1", opts.paramName)
+			}
+
+			err = opts.Validate()
+			if err != nil {
+				t.Errorf("Expected nil error, got %s", err)
+				return
+			}
+
+			if tt.exists || tt.forceFlag {
+				prefClient.EXPECT().DeleteConfiguration("arg1")
```

Comment:
Should this return any error, or if nothing is mentioned, it returns a default value?

Reply:
I cannot find any documentation on this, but I assume it will return the zero values for all the returned values if we don't specify any other value.